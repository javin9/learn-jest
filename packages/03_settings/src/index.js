const add = (a, b) => {
  return a + b;
};

const minus = (a, b) => {
  return a - b;
};

const multiply = (a, b) => {
  return a * b;
};

module.exports = {
  add,
  minus,
  multiply,
};
