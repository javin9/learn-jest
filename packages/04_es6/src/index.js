export const add = (a, b) => {
  return a + b;
};

export const minus = (a, b) => {
  return a - b;
};

export const multiply = (a, b) => {
  return a * b;
};


