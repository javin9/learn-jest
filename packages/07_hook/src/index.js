export class Counter {
  constructor() {
    this.num = 1;
  }

  // add 加法
  add() {
    this.num += 1;
  }

  addTwo() {
    this.num += 2;
  }

  // minus 减法
  minus() {
    this.num -= 1;
  }

  minusTwo() {
    this.num -= 2;
  }
}
