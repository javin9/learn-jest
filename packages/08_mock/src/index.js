export const runCallback = (callback) => {
  callback();
};

export const runCallbackParams = (callback) => {
  callback('冰箱太凉');
};

export const createNewObject = (ClassName) => {
  return new ClassName();
};
