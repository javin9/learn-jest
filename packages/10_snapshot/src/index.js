// 对配置文件测试
export const getConfig = () => {
  return {
    server: 'https://localhost',
    port: 8082,
  };
};

export const getCommonConfig = () => {
  return {
    db: 'mongodb://10.90.73.48:27097/lowcodedb',
  };
};

export const getRandomConfig = () => {
  return {
    name: '冰箱太凉',
    time: new Date(),
  };
};
