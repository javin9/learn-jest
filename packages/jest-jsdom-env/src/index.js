exports.isCurrentHost = function (host) {
  return window.location.host == host;
};
