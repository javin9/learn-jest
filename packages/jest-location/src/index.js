import { clearLocation, fakeLocation } from './location'

import { clearUserAgent, fakeUserAgent } from './userAgent'

export { fakeLocation, clearLocation, clearUserAgent, fakeUserAgent }
