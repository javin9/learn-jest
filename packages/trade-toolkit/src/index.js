/**
 * 公共导出入口
 */

console.log('index module')
