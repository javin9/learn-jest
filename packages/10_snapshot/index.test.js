import { getConfig, getCommonConfig, getRandomConfig } from './src/index';

// test('测试getConfig配置文件', () => {
//   expect(getConfig()).toEqual({
//     server: 'https://localhost',
//     port: 8080,
//   });
// });

// -----------------------------------
// 和快照匹配
// 第一次调用的时候，会把getConfig的结果 自动生成一个快照
// 第二次 再次生成一个快照，和之前的进行匹配 ,只要改了配置文件，就会报错
// 如何覆盖快照 w+u
test('测试getConfig 和快照匹配', () => {
  expect(getConfig()).toMatchSnapshot();
});

// // -----------------------------------
// // 两个快照如何测试
// // w+i 交互式 ，只提示一个
// test('测试getCommonConfig 和快照匹配', () => {
//   expect(getCommonConfig()).toMatchSnapshot();
// });

// -----------------------------------
// 快照里面time 只要是个Date类型都的任何时间都可以
// test('变化的配置', () => {
//   expect(getRandomConfig()).toMatchSnapshot({ time: expect.any(Date) });
// });

// ----------------------------
// test('行内快照', () => {
//   expect(getConfig()).toMatchInlineSnapshot();
// });
