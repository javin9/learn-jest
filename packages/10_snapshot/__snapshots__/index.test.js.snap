// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试getConfig 和快照匹配 1`] = `
{
  "port": 8080,
  "server": "https://localhost",
}
`;
